pkg/operator/controller/externaldns/deployment.go

```diff
@@ -127,7 +154,7 @@ func (r *reconciler) currentExternalDNSDeployment(ctx context.Context, nsName ty
 }
 
 // desiredExternalDNSDeployment returns the desired deployment resource.
-func desiredExternalDNSDeployment(namespace, image, secretName string,
+func desiredExternalDNSDeployment(namespace, image, secretName string, secretHash string,
```

Comment:
wow 9 parameters - should you consider using a new struct ? its not a big deal, usually in go with more than 4 params you should look at creating a new struct

Reply:
Done here ac35fea8394353bf3ae40e85fc716e258b203025